Source/Core/Core/PowerPC/JitCommon/JitCache.cpp

```diff
@@ -324,9 +324,20 @@ using namespace Gen;
 	{
 		XEmitter emit(location);
 		if (*location == 0xE8)
+		{
 			emit.CALL(address);
+		}
 		else
-			emit.JMP(address, true);
+		{
+			// If we're going to link with the next block, there is no need
+			// to emit JMP. So just NOP out the gap to the next block.
+			// Support up to 3 additional bytes because of alignment.
+			s64 offset = address - emit.GetCodePtr();
+			if (offset > 0 && offset <= 5 + 3)
```

Comment:
It's just confusing to me that you check for a condition that is never going to happen and doesn't make sense anyway (infinite loop or jumping into the jmp).